src/transaction/log_prior_recv.cpp

```diff
@@ -58,11 +59,50 @@ namespace cublog
     std::unique_lock<std::mutex> ulock (m_messages_mutex);
     m_shutdown = true;
     ulock.unlock ();
-    m_messages_condvar.notify_one ();
+    m_messages_push_cv.notify_one ();
 
     m_thread.join ();
   }
 
+  void
+  prior_recver::wait_until_empty_and_pause ()
+  {
+    std::unique_lock<std::mutex> ulock (m_messages_mutex);
+
+    assert (m_paused == false);
+    assert (m_shutdown == false);
+
+    constexpr auto sleep_30ms = std::chrono::milliseconds (30);
+    while (!m_messages_consume_cv.wait_for (ulock, sleep_30ms, [this]
+    {
+      return m_messages.empty();
+      }));
+
+    m_paused = true;
+  }
+
+  void
+  prior_recver::resume ()
+  {
+    std::unique_lock<std::mutex> ulock (m_messages_mutex);
```

Comment:
does `m_message_mutex` also protect the `m_paused` variable?  
Maybe we can remove `ulock ` or unlock it before notify_one ().